contributors/design-proposals/customresources-subresources.md

```diff
@@ -0,0 +1,221 @@
+
+### API Types
+
+The addition of the following types in `apiextensions.k8s.io/v1beta1` is proposed:
+
+```go
+type CustomResourceDefinitionSpec struct {
+    Group string
+    Version string
+    Names CustomResourceDefintion
+    Scope ResourceScope
+    Validation *CustomResourceValidation
+    // SubResources describes the subresources for CustomResources
+    // This field is alpha-level and should only be sent to servers that enable
+    // SubResources via the CurstomResourceSubResources feature gate.
+    // +optional
+    SubResources *CustomResourceSubResources `json:“subResources,omitempty”`
+}
+
+// CustomResourceSubResources defines the status and scale sub-resources for CustomResources.
+type CustomResourceSubResources {
+    // Status denotes the status subresource for CustomResources
+    Status *CustomResourceSubResourceStatus `json:“status,omitempty”`
+    // Scale denotes the scale subresource for CustomResources
+    Scale  *CustomResourceSubResourceScale  `json:“scale,omitempty”`
+}
+
+// CustomResourceSubResourceStatus defines how to serve the HTTP path <CR Name>/status.
+type CustomResourceSubResourceStatus struct {
+    // The JSON path (e.g. “.status”) of the status of a CustomResource.
+    // The whole object is transferred over the wire, but only the status can be mutated via the /status subresource.
+    // StatusPath is restricted to be “.status” and defaults to “.status”.
+    StatusPath string `json:“path,omitempty”`
+    // The JSON path (e.g. “.spec”) of the spec of a CustomResource.
+    // SpecPath is restricted to be “.spec” and defaults to “.spec”.
+    // Changes to the specified JSON path increase the “.metadata.generation” value.
+    SpecPath string `json:“path,omitempty”`
+}
+
+// CustomResourceSubResourceScale defines how to serve the HTTP path <CR name>/scale.
+type CustomResourceSubResourceScale struct {
+    // required, e.g. “.spec.replicas”.
+    // Only JSON paths without the array notation are allowed.
+    SpecReplicasPath string `json:“specReplicasPath,omitempty”`
+    // optional, e.g. “.status.replicas”.
+    // Only JSON paths without the array notation are allowed.
+    StatusReplicasPath string `json:“statusReplicasPath,omitempty”`
+    // optional, e.g. “.spec.labelSelector”.
+    // Only JSON paths without the array notation are allowed.
+    LabelSelectorPath string `json:“labelSelectorPath,omitempty”`
+}
+
+// The following is the payload to send over the wire for /scale. It happens
+// to be defined here in apiextensions.k8s.io because we don’t have a global
+// Scale type in meta/v1. Ref: https://github.com/kubernetes/kubernetes/issues/49504
+
+// Scale represents a scaling request for a resource.
+type Scale struct {
+    metav1.TypeMeta `json:",inline"`
+    // Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata.
+    // +optional
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
protobuf tags are missing for a lot of fields. I would drop them from the proposal and add them later in code.